woocommerce/woocommerce-android
# Add Tracks Events for Linked Products

Closes #3159. 

This primarily adds tracking for Linked Products. At the same time, this also replaces the existing `GROUPED_PRODUCT_LINKED_PRODUCTS_*` events so they are all using just one event, `CONNECTED_PRODUCTS_LIST`, but with different `context` values. This change was discussed in https://github.com/woocommerce/woocommerce-ios/issues/3151. 

I also checked the `GROUPED_PRODUCT_LINKED_PRODUCTS_*` events previously and it didn't look like we were using them in Funnels so it should be fine to replace them. 

The events are the same as iOS (https://github.com/woocommerce/woocommerce-ios/pull/3425/files). 

## Testing

1. Navigate to a Product → Linked Products. 
2. Confirm this event is triggered: `linked_products, Properties: {"action":"shown", ...}`
3. Tap on Add or Edit for Upsells. 
4. Tap on Add Product. 
5. Confirm this event is tracked: `connected_products_list, Properties: {"context":"upsells","action":"add_tapped", ...}`.
6. Select a product. Tap Done.
7. Confirm this event is tracked: `connected_products_list, Properties: {"context":"upsells","action":"added", ...}`. 
8. Delete the added product. 
9. Confirm this event is tracked: `connected_products_list, Properties: {"context":"upsells","action":"delete_tapped", ...}`
10. Add another product and tap Done.
11. Tap on Done in the Upsells list. 
12. Confirm this event is tracked: `connected_products_list, Properties: {"context":"upsells","action":"done_tapped", ...}`
13. You should be navigated back to the Linked Products screen. Tap Done. 
14. Confirm this event is tracked: `linked_products, Properties: {"action":"done", ...}`
15. Repeat steps 3 to 12 for Cross-sells and confirm the same events but the `context` should be `cross_sells`. 
16. Repeat steps 3 to 12 for editing a Grouped Product's linked product and confirm the same events but the `context` should be `grouped_products`. 

## Submitter Checklist
